pkg/streamingpromql/functions/instant_vector.go

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package functions
+
+import (
+	"math"
+
+	"github.com/prometheus/prometheus/promql"
+
+	"github.com/grafana/mimir/pkg/streamingpromql/pooling"
+	"github.com/grafana/mimir/pkg/streamingpromql/types"
+)
+
+type SeriesMetadataFunction func(seriesMetadata []types.SeriesMetadata, pool *pooling.LimitingPool) ([]types.SeriesMetadata, error)
+
+func DropSeriesName(seriesMetadata []types.SeriesMetadata, _ *pooling.LimitingPool) ([]types.SeriesMetadata, error) {
+	for i := range seriesMetadata {
+		seriesMetadata[i].Labels = seriesMetadata[i].Labels.DropMetricName()
+	}
+
+	return seriesMetadata, nil
+}
+
+type InstantVectorFunction func(seriesData types.InstantVectorSeriesData, pool *pooling.LimitingPool) (types.InstantVectorSeriesData, error)
+
+func TransformationFunc(transform func(f float64) float64) InstantVectorFunction {
```

Comment:
For these functions, what do we do with native histograms if they're present? Should they be dropped? (If so, we should return the `HPoint` slice to the pool.)